### PR TITLE
Add marker for case insensitive test collection by component and importance

### PR DESCRIPTION
**Pull request by shweta83**
### Problem Statement
We need case insensitive test collection component-wise

### Solution
Added marker for component-wise test collection
result:
pytest tests/foreman/ --component auditlog
collected 5434 items / 5426 deselected / 8 selected                                                                                              

```
tests/foreman/api/test_audit.py ✓✓✓                                                                                                38% ███▊      
 tests/foreman/ui/test_audit.py ✓✓✓✓✓                                                                                              100% ██████████
```
=======================================================

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ --component alternatecontentsources --collect-only

**Comment by shweta83** [comment]
pytest tests/foreman/  --component auditlog

File: pytest_plugins/metadata_markers.py [hunk 1]
```diff
@@ -191,8 +191,8 @@ def pytest_collection_modifyitems(items, config):
             # Add marker starting at smallest docstring scope
             # only add the mark if it hasn't already been applied at a lower scope
             doc_component = component_regex.findall(docstring)
-            if doc_component and 'component' not in item_mark_names:
-                item.add_marker(pytest.mark.component(doc_component[0]))
+            if doc_component and 'CaseComponent' not in item_mark_names:
```
Comment (omkarkhatavkar) [inline pytest_plugins/metadata_markers.py:194]: why did you change the `component` to `CaseComponent`. we are adding `component` as marker to the test not `CaseComponent`. Or did I missed anything here ?

**Comment by jyejare** [comment]
@shweta83 Please revisit the tests are failing.

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: pytest tests/foreman/ --component auditlog

**Comment by shweta83** [comment]
> @shweta83 Please revisit the tests are not collecting. Please let me know if u need the help.

It was working well on my local setup. I need to check what's the issue running on CI.

File: pytest_plugins/metadata_markers.py (hunk 1 above)
Comment (shweta83) [reply pytest_plugins/metadata_markers.py:194]: It searches for CaseComponent docstring in the test.

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ --component auditlog

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6919
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ --component auditlog --external-logging
Test Result : ======== 8 passed, 5442 deselected, 5563 warnings in 1140.10s (0:19:00) ========
```

**Comment by shweta83** [comment]
> @shweta83 Please revisit the tests are not collecting. Please let me know if u need the help.

Tests are collected as expected. I messed up in running previous PRT which caused the test collection failed.

File: pytest_plugins/metadata_markers.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def pytest_collection_modifyitems(items, config):
     # split the option string and handle no option, single option, multiple
     # config.getoption(default) doesn't work like you think it does, hence or ''
     importance = [i for i in (config.getoption('importance') or '').split(',') if i != '']
```
Comment (rplevka) [inline pytest_plugins/metadata_markers.py:166]: Mind doing the same to importance, while at it?
```suggestion
    importance = [i.lower() for i in (config.getoption('importance') or '').split(',') if i != '']
```
+ update the commit message

**Review by rplevka — APPROVED** [review]

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ --component auditlog

File: pytest_plugins/metadata_markers.py (hunk 1 above)
Comment (omkarkhatavkar) [reply pytest_plugins/metadata_markers.py:194]: @jyejare I think it should be `component` right ? `CaseComponent` would not be a right marker to check as it it will not be there. As the we are adding the component marker.

Comment (jyejare) [reply pytest_plugins/metadata_markers.py:194]: @shweta83 Here @omkarkhatavkar is right we are adding a `keypair` of `component=<somcomponent>` to the the test item.

In below `compoenent not in item_mark_names` verifies that the component marker is not already added to the item.
```
if doc_component and 'component' not in item_mark_names:
```

File: pytest_plugins/metadata_markers.py [hunk 3]
```diff
@@ -191,8 +191,8 @@ def pytest_collection_modifyitems(items, config):
             # Add marker starting at smallest docstring scope
             # only add the mark if it hasn't already been applied at a lower scope
             doc_component = component_regex.findall(docstring)
-            if doc_component and 'component' not in item_mark_names:
-                item.add_marker(pytest.mark.component(doc_component[0]))
+            if doc_component and 'CaseComponent' not in item_mark_names:
+                item.add_marker(pytest.mark.component(doc_component[0].lower()))
             doc_importance = importance_regex.findall(docstring)
             if doc_importance and 'importance' not in item_mark_names:
                 item.add_marker(pytest.mark.importance(doc_importance[0]))
```
Comment (jyejare) [inline pytest_plugins/metadata_markers.py:198]:
```suggestion
                item.add_marker(pytest.mark.importance(doc_importance[0].lower()))
```

**Review by jyejare — CHANGES_REQUESTED** [review]
ACK pending comments.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6932
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ --component auditlog --external-logging
Test Result : ======== 8 passed, 5445 deselected, 5563 warnings in 900.26s (0:15:00) =========
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ --component auditlog

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6982
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ --component auditlog --external-logging
Test Result : ======== 8 passed, 5461 deselected, 5583 warnings in 1092.40s (0:18:12) ========
```

**Review by jyejare — APPROVED** [review]
